param/parameterized.py

```diff
@@ -411,6 +411,39 @@ def get_method_owner(method):
     return method.__self__
 
 
+# Hooks to apply to depends and bind arguments to turn them into valid parameters
+DEPENDENCY_TRANSFORMS = []
```

Comment:
I don't think this should be capitalized, as it is not a constant.

I could be a bit scared that someone overwrites this with his own list of transforms, making things fail in Panel and hard to debug it. 

Some suggestions could be to have an API for adding hooks. Maybe even register them in a dictionary where the key is the package name, and the values are a list of transforms. Maybe even have a way only to choose the dependencies from a specific package.

Reply:
Indeed, an API for adding hooks seems sensible.